cmd/cluster/validatepullsecret.go

```diff
@@ -11,93 +17,233 @@ import (
+}
+
+// getPullSecretFromCluster runs a managed-script to get the pull-secret
+// email from cluster without backplane elevation
+// it returns the email, error and done
+// done means a service log has been sent
+func getPullSecretFromCluster(clusterID string) (string, error, bool) {
+	fmt.Println("Creating a managedjob to get pull-secret email in the cluster")
+	createJobCmd := "ocm backplane managedjob create SREP/get-pull-secret-email"
+	createJobOutput, err := exec.Command("bash", "-c", createJobCmd).CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to run managed script SREP/get-pull-secret-email:\n%s", strings.TrimSpace(string(createJobOutput))), false
+	}
+	// Get the job id from output
+	re := regexp.MustCompile(`openshift-job-[a-z0-9]+`)
+	matches := re.FindStringSubmatch(string(createJobOutput))
+	if len(matches) == 0 {
+		return "", fmt.Errorf("failed to find job id after creating managedjob"), false
+	}
+	jobId := matches[0]
+	fmt.Printf("managedjob id: %s\n", jobId)
+
+	fmt.Println("Waiting for managedjob to finish")
+	getJobStatusCmd := fmt.Sprintf("ocm backplane managedjob get %s", jobId)
+	re = regexp.MustCompile(`Succeeded`)
+	// wait 10x10 seconds to finish
+	for i := 0; i < 10; i++ {
+		getJobStatusOutput, err := exec.Command("bash", "-c", getJobStatusCmd).CombinedOutput()
+		if err != nil {
+			continue
+		}
+		matches = re.FindStringSubmatch(string(getJobStatusOutput))
+		if len(matches) > 0 {
+			break
+		}
+		time.Sleep(10 * time.Second)
+	}
+	// managedjob timeout, check the error.
+	if len(matches) == 0 {
+		// instead of the go native way to get events:
+		// https://github.com/openshift/oc/blob/c7b582ed27cfb2890068d6cb29cb2f5b936654cd/vendor/k8s.io/kubectl/pkg/describe/describe.go#L4258
+		// we can simply run the oc describe command and match the output.
+		describeJobCmd := fmt.Sprintf("oc describe pod %s -n openshift-backplane-managed-scripts", jobId)
+		describeJobOutput, err := exec.Command("bash", "-c", describeJobCmd).CombinedOutput()
+		if err != nil {
+			return "", fmt.Errorf("failed to execute: %s: %w", describeJobCmd, err), false
+		}
+		re = regexp.MustCompile(`unauthorized`)
+		matches = re.FindStringSubmatch(string(describeJobOutput))
+		if len(matches) > 0 {
+			fmt.Printf("managedjob failed due to failed authentication to pull image, run the below command for more detail:\n%s\n", describeJobCmd)
+			fmt.Println("Sending service log")
+			postCmd := servicelog.PostCmdOptions{
+				Template:  "https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/pull_secret_change_breaking_upgradesync.json",
+				ClusterId: clusterID,
+			}
+			if err = postCmd.Run(); err != nil {
+				return "", err, true
+			}
+			return "", nil, true
+		}
+		return "", fmt.Errorf("managedjob timeout, try --elevate to validate pull-secret without a managed job"), false
+	}
+
+	// Get the output of the managed script
+	getJobResultCmd := fmt.Sprintf("ocm backplane managedjob logs %s", jobId)
```

Comment:
I like the interaction with the backplane CLI, probably we could even use it as package instead of a shell-out, that would be great. Probably not here if that's a bigger thing, but I feel like we may want to add more such interactions in the future.

Reply:
At least I think the interactions with backplane managed scripts should be wrapped, so they don't clutter this function too much.

Reply:
Yeah, having some library instead of direct shell call would make it more maintainable.  For this time I think we can use the shell call to deliver a MVP and enhance in the future.

I split the managedjob related operation to smaller functions in the new commit.